cmd/kube-bench-summarizer/summarizer/summarizer.go

```
@@ -0,0 +1,200 @@
+							ID:     group.ID,
+							Text:   group.Text,
+							Checks: []*CheckWrapper{},
+						}
+						s.groupsMap[group.ID] = g
+						s.summarizedReport.Groups = append(s.summarizedReport.Groups, g)
+					}
+
+					c := s.checksMaps[check.ID]
+					if c == nil {
+						logrus.Infof("check not found in map: %v", check.ID)
+						c = &CheckWrapper{
+							ID:          check.ID,
+							Text:        check.Text,
+							Remediation: check.Remediation,
+							TestInfo:    check.TestInfo,
+							State:       check.State,
+							Hosts:       []string{},
+						}
+						s.checksMaps[check.ID] = c
+						g.Checks = append(g.Checks, c)
+
+						switch check.State {
+						case kb.FAIL:
+							g.Fail++
+							s.summarizedReport.Fail++
+						case kb.WARN:
+							g.Warn++
+							s.summarizedReport.Warn++
+						case kb.INFO:
+							g.Info++
+							s.summarizedReport.Info++
+						}
+					}
+					c.Hosts = append(c.Hosts, hostname)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (s *Summarizer) save() error {
+	data, err := json.MarshalIndent(s.summarizedReport, "", " ")
+	if err != nil {
+		logrus.Errorf("error marshaling: %v", err)
+		return err
+	}
+
+	outputFilePath := fmt.Sprintf("%s/%s", s.OutputDirectory, DEFAULT_OUTPUT_FILE_NAME)
+	err = ioutil.WriteFile(outputFilePath, data, 0644)
+	if err != nil {
+		logrus.Errorf("error writing report file: %v", err)
+		return err
+	}
+
+	return nil
+}
+
+func (s *Summarizer) Summarize() error {
+	logrus.Infof("summarize")
+	logrus.Debugf("inputDir: %v", s.InputDirectory)
+
+	// Walk through the host folders
+	hostsDir, err := ioutil.ReadDir(s.InputDirectory)
+	if err != nil {
+		logrus.Errorf("error listing directory: %v", err)
+		return err
+	}
+
+	for _, hostDir := range hostsDir {
+		if !hostDir.IsDir() {
+			continue
+		}
+		hostname := hostDir.Name()
+		logrus.Debugf("hostDir: %v", hostname)
+
+		if err := s.summarizeForHost(hostname); err != nil {
+			logrus.Errorf("error summarizeForHost: %v", hostname)
```

do we want to just log the error and proceed? I am thinking of a case when the summarizer would fail, but rancher gets no error back.

---

Agree with you, I think we should return an error so that the problem can be debugged.